make/default_compiler_options

```diff
@@ -12,8 +12,24 @@ AR = ar
 # CPPFLAGS are used for both C and C++ compilation
 CPPFLAGS = -DNO_FPRINTF_OUTPUT -pipe
 # CXXFLAGS are just used for C++
-CXXFLAGS = -Wall -I . -isystem $(EIGEN) -isystem $(BOOST) -isystem $(CVODES)/include -std=c++1y -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION -Wno-unused-function -Wno-uninitialized
+CXXFLAGS = -Wall -I . -isystem $(EIGEN) -isystem $(BOOST) -isystem $(CVODES)/include -isystem $(OPENCL) -std=c++1y -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION -Wno-unused-function -Wno-uninitialized -DSTAN_GPU -DSTAN_DEVICE_CPU
+
 GTEST_CXXFLAGS = -DGTEST_USE_OWN_TR1_TUPLE
 LDLIBS =
 EXE =
 WINE =
+POST_LDLIBS =
+
+ifeq ($(shell echo $(TRAVIS)),true)
+  ifeq ($(shell echo $(TRAVIS_OS_NAME)),linux)
+    POST_LDLIBS+=-L/home/travis/AMDAPPSDK/lib/x86_64/
+  endif
+endif
+
+ifneq (,$(findstring STAN_GPU, $(CXXFLAGS)))
+  ifeq ($(OS_TYPE),Darwin)
+    POST_LDLIBS+=-framework OpenCL
+  else
+    POST_LDLIBS+=-lOpenCL
+  endif
+endif
```

Comment:
Relevant side note: I have no idea how to compile this for windows. Does Jenkins test on Windows?

Reply:
Jenkins does test on Windows.